internal/bed/merging.go

```diff
@@ -0,0 +1,61 @@
+package bed
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// Merge lines in bed file
+func (bf *Bedfile) MergeLines() {
+	var merged Line
+	var mergedLines []Line
+	for i, l := range mergeSort(bf.Lines) {
+		// If the lines are overlapping or touching merge them
+		if i != 0 &&
+			merged.Chr == l.Chr &&
+			merged.Strand == l.Strand &&
+			merged.Feat == l.Feat &&
+			merged.Stop+bf.Overlap >= l.Start-1 {
+			// Set new stop if it is later than the
+			// merged stop
+			if l.Stop > merged.Stop {
+				merged.Stop = l.Stop
+				merged.Full[stopIdx] = strconv.Itoa(l.Stop)
+			}
+			// Join information in the optional columns
+			if len(l.Full) > stopIdx {
```

Comment:
I am just thinking that `stopIdx` is 2 and if `l.Full` will give you 3 if you have only three columns but I would consider that not a case where we join information